pkg/server/server.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+	"strconv"
+
+	camelv1alpha "github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
+	"github.com/gin-gonic/gin"
+	"github.com/sco1237896/sco-backend/pkg/client"
+)
+
+var (
+	logger = slog.Default()
+	cl     client.Client
+)
+
+type Options struct {
+	Addr string
+	Port int
+}
+
+func Start(opts Options, client *client.Client) error {
+	r := setupRouter(client)
+	logger.Info("starting server")
+	err := r.Run(opts.Addr + ":" + strconv.Itoa(opts.Port))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupRouter(client *client.Client) *gin.Engine {
+	cl = *client
+	r := gin.Default()
+	r.GET("/pipes", getPipes)
```

Comment:
instead of storing the client globally, you can use something like 

```go
r.GET("/pipes", func(c *gin.Context) { 
    getPipes(client, c) 
})
```

Reply:
https://github.com/sco1237896/sco-backend/pull/1/commits/0ad411bff9c78c47d4e93a88a1f923b35002213f